scripts/travis_test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+#
+# This file is part of clr-boot-manager.
+#
+# Copyright © 2017 Ikey Doherty
+#
+# clr-boot-manager is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1
+# of the License, or (at your option) any later version.
+#
+set -e
+
+testRoot=`pwd`
+suppressions="$testRoot/sgcheck.suppressions"
+coverDir="$testRoot/outCoverage"
+
+# Nuke any existing builds
+function nuke_build() {
+    if [[ -d "build" ]]; then
+        rm -rf "build"
+    fi
+}
+
+# Just perform a single build
+function build_one() {
+    meson build --buildtype debugoptimized -Db_coverage=true --prefix=/usr --sysconfdir=/etc --datadir=/usr/share -Dwith-systemd-system-unit-dir=/lib/systemd/system $*
+    ninja -C build
+}
+
+# Do a check with the various valgrind tools
+function check_valgrind() {
+    local valgrindArgs="valgrind --suppressions=\"$suppressions\" --error-exitcode=1"
+
+    # Memory test
+    meson test -C build --print-errorlogs --logbase=memcheck --wrap="$valgrindArgs --tool=memcheck --leak-check=full --show-reachable=no"
+    meson test -C build --print-errorlogs --logbase=helgrind --wrap="$valgrindArgs --tool=helgrind"
+    meson test -C build --print-errorlogs --logbase=drd --wrap="$valgrindArgs --tool=drd"
+    # meson test -C build --logbase=sgcheck --wrap="$valgrindArgs --tool=exp-sgcheck"
+}
+
+# Do a "normal" test suite check
+function check_normal() {
+    meson test -C build --print-errorlogs
+}
+
+# Store the coverage report. If we have one, merge the new report. Finally,
+# strip any unneeded noise from the report, to prepare it for upload
+function stash_coverage() {
+    ninja -C build coverage-html
+    local coverageFile="$coverDir/coverage.info"
+    local sampleFile="./build/meson-logs/coverage.info"
+
+    if [[ ! -d "$coverDir" ]]; then
+        mkdir "$coverDir"
+    fi
+
+    # Does this guy exist?
+    if [[ ! -e "$coverageFile" ]]; then
+        cp -v "$sampleFile" "$coverageFile"
+    else
+        # Merge them!
+        lcov -a "$coverageFile" -a "$sampleFile" -o "${coverageFile}.tmp"
+
+        # Stick this guy back as the main coverage file now
+        mv "${coverageFile}.tmp" "$coverageFile"
+    fi
+
+    # Ensure we remove any unnecessary junk now
+    lcov --remove "$coverageFile" 'tests/*' '/usr/*' --output-file "$coverageFile"
+}
+
+# Ensure we're protected against any potential glib infractions
```

Comment:
Not needed here right?

Reply:
It was mostly about reproducing the functionality of the ax valgrind macro (come to think of it I need to nuke the m4 tree) in case we ever end up linking to glib crap, but I can remove it.

Reply:
Yea, I'd assume bad times if we start to link in glib stuff but I suppose we could figure out a way to split it off to be not required by the main components as we want to keep our dependencies tiny.

Reply:
Yeah, I've dropped it from the PR - I'll check again but should be gone now?